File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.Stack;

class Solution
{
    public int solution(String s)
    {
        int answer = 0;
        Stack<Character> st = new Stack<>();
        
        for(int i=0; i<s.length(); i++){
            st.add(s.charAt(i));
            
            if(st.size() >= 2){
                char top = st.peek();
                char next = st.elementAt(st.size()-2);
                
                if(top == next){
                    st.pop();
                    st.pop();
                }
            }
        }

        if(st.size() == 0){
            answer = 1;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = -1

    stack = []
    
    for i in range(len(s)):
        stack.append(s[i])
        
        if len(stack) >= 2:
            top1, top2 = stack[-1], stack[-2]
            if top1 == top2:
                stack.pop()
                stack.pop()
            
    answer = 1 if len(stack) == 0 else 0

    return answer